src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1541,1900 @@ interface MPSMatrixMultiplication {
+		nuint KernelHeight { get; set; }
+
+		[Static]
+		[Export ("nodeWithSource:kernelSize:")]
+		MPSCnnSpatialNormalizationNode Create (MPSNnImageNode sourceNode, nuint kernelSize);
+
+		[Export ("initWithSource:kernelSize:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (MPSNnImageNode sourceNode, nuint kernelSize);
+
+		[Export ("initWithSource:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (MPSNnImageNode sourceNode);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSCnnNormalizationNode), Name = "MPSCNNLocalContrastNormalizationNode")]
+	interface MPSCnnLocalContrastNormalizationNode
+	{
+		[Export ("pm")]
+		float Pm { get; set; }
+
+		[Export ("ps")]
+		float Ps { get; set; }
+
+		[Export ("p0")]
+		float P0 { get; set; }
+
+		[Export ("kernelWidth")]
+		nuint KernelWidth { get; set; }
+
+		[Export ("kernelHeight")]
+		nuint KernelHeight { get; set; }
+
+		[Static]
+		[Export ("nodeWithSource:kernelSize:")]
+		MPSCnnLocalContrastNormalizationNode Create (MPSNnImageNode sourceNode, nuint kernelSize);
+
+		[Export ("initWithSource:kernelSize:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (MPSNnImageNode sourceNode, nuint kernelSize);
+
+		[Export ("initWithSource:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (MPSNnImageNode sourceNode);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSCnnNormalizationNode), Name = "MPSCNNCrossChannelNormalizationNode")]
+	interface MPSCnnCrossChannelNormalizationNode
+	{
+		[Export ("kernelSizeInFeatureChannels")]
+		nuint KernelSizeInFeatureChannels { get; set; }
+
+		[Static]
+		[Export ("nodeWithSource:kernelSize:")]
+		MPSCnnCrossChannelNormalizationNode Create (MPSNnImageNode sourceNode, nuint kernelSize);
+
+		[Export ("initWithSource:kernelSize:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (MPSNnImageNode sourceNode, nuint kernelSize);
+
+		[Export ("initWithSource:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (MPSNnImageNode sourceNode);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSNnFilterNode), Name = "MPSNNScaleNode")]
+	interface MPSNnScaleNode
+	{
+		[Static]
+		[Export ("nodeWithSource:outputSize:")]
+		MPSNnScaleNode Create (MPSNnImageNode sourceNode, MTLSize size);
+
+		[Static]
+		[Export ("nodeWithSource:transformProvider:outputSize:")]
+		MPSNnScaleNode Create (MPSNnImageNode sourceNode, [NullAllowed] IMPSImageTransformProvider transformProvider, MTLSize size);
+
+		[Export ("initWithSource:outputSize:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, MTLSize size);
+
+		[Export ("initWithSource:transformProvider:outputSize:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, [NullAllowed] IMPSImageTransformProvider transformProvider, MTLSize size);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSNnFilterNode), Name = "MPSNNBinaryArithmeticNode")]
+	interface MPSNnBinaryArithmeticNode
```

Comment:
unless there are subclasses, this looks immutable so I would guess init (default constructor) does not do any good -> [DisableDefaultCtor]

Also Nn -> NN

Reply:
This does have several subclasses, albeit all empty atm:

````        [TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
	[BaseType (typeof(MPSNNBinaryArithmeticNode))]
	interface MPSNNAdditionNode
	{
	}````

On that note, I just realized these probably need to re-expose the actors and create methods from the MPSBinaryArithmeticNode with the subtypes  don't they?